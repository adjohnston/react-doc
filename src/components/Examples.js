import React, { Component } from 'react'
import PropTypes from 'prop-types'
import className from 'helpers/class-name'

class Examples extends Component {
  constructor (props) {
    super(props)

    this.state = {
      current: 'default'
    }

    this.onChange = this.onChange.bind(this)
  }

  onChange ({target: { value }}) {
    this.props.setExample(value)
    this.setState({current: value})
  }

  render () {
    const examples = this.props.examples

    return (examples && (
      <section>
        <h2
          className={className('title')}>
          Examples
        </h2>

        <div
          className={className('dropdown')}>
          <select
            className={className('dropdown__select')}
            onChange={this.onChange}
            value={this.state.current}>
            {Object.keys(examples).map(example => (
              <option
                key={example}>
                {example}
              </option>
            ))}
          </select>

          <div className={className('dropdown__arrow')} />
        </div>
      </section>
    )) || null
  }
}

Examples.propTypes = {
  setExample: PropTypes.func.isRequired,
  examples: PropTypes.object
}

export default Examples
