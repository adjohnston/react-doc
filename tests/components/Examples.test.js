import React from 'react'
import renderer from 'react-test-renderer'
import Examples from 'src/components/Examples'

test('it should not render without examples', () => {
  const props = {
    setExample: () => {}
  }

  const tree = renderer
    .create(<Examples {...props} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it should render correctly', () => {
  const props = {
    examples: {},
    setExample: () => {}
  }

  const tree = renderer
    .create(<Examples {...props} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

test('it should show the value when the selection is changed', () => {
  const props = {
    examples: {
      'something else': {}
    },
    setExample: () => {}
  }

  const component = renderer.create(<Examples {...props} />)

  component
    .getInstance()
    .onChange({target: {value: 'something else'}})

  expect(component.toJSON()).toMatchSnapshot()
})
