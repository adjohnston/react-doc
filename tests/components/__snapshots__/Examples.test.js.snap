// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should not render without examples 1`] = `null`;

exports[`it should render correctly 1`] = `
<section>
  <h2
    className="stncl__title"
  >
    Examples
  </h2>
  <div
    className="stncl__dropdown"
  >
    <select
      className="stncl__dropdown__select"
      onChange={[Function]}
      value="default"
    />
    <div
      className="stncl__dropdown__arrow"
    />
  </div>
</section>
`;

exports[`it should show the value when the selection is changed 1`] = `
<section>
  <h2
    className="stncl__title"
  >
    Examples
  </h2>
  <div
    className="stncl__dropdown"
  >
    <select
      className="stncl__dropdown__select"
      onChange={[Function]}
      value="something else"
    >
      <option>
        something else
      </option>
    </select>
    <div
      className="stncl__dropdown__arrow"
    />
  </div>
</section>
`;
